refactor: port to gtk4

ChildishGiant opened the pull request:
Currently a WIP

ChildishGiant committed:
refactor: port to gtk4

ChildishGiant committed:
build: :construction_worker: Change build instructions to use flatpak-builder

ChildishGiant committed:
wip: start moving over to gnome

ChildishGiant committed:
wip: lots of progress to gnome styles

ChildishGiant committed:
WIP: Whatever I did ages ago

ChildishGiant committed:
wip: progress on gtk4/gnome port

fixed issues with several apps showing up along side the no apps status

ChildishGiant committed:
Update ci.yml

change runtime to gnome and updated flatpak-builder version

ChildishGiant committed:
ci: :green_heart: Fix some linter errors and removed old borked screenshots to hopefully let CI run

ChildishGiant committed:
Merge branch 'gtk4' of https://github.com/childishgiant/mixer into gtk4

ChildishGiant committed:
ci: 💚 Fix some daft issues causing CI to stumble

ChildishGiant commented:
@ryonakano I've made some changes in this branch that clash with the PR you submitted a bit ago to fix language building. I don't really understand the arcane magic of meson so would love if you had the time to look at this. The CI build is only failing due to appstream not liking the old screenshot dimensions, locally building works fine

ryonakano commented on src/application.vala:
```diff
@@ -0,0 +1,109 @@
+        }
+
+
+        construct {
```
You need the following code somewhere in the constructor of this class to show your app in the user language:

```vala
            Intl.setlocale (LocaleCategory.ALL, "");
            Intl.bindtextdomain (Constants.GETTEXT_PACKAGE, Constants.LOCALE_DIR);
            Intl.bind_textdomain_codeset (Constants.GETTEXT_PACKAGE, "UTF-8");
            Intl.textdomain (Constants.GETTEXT_PACKAGE);
```

ryonakano reviewed (commented):
It looks like the meson things are correctly set up, just we're missing some code in the Applications class.

ChildishGiant committed:
fix: convert some strokes in the icon to paths to hopefully fix it not displaying correctly in certain places.

ChildishGiant committed:
fix: remove drop shadow on icon as this is no longer good practice

ChildishGiant committed:
update translation files + build system

ChildishGiant committed:
wip: try to save the user's window size/position
"Settings schema 'com.github.childishgiant.mixer' is not installed" I think flatpak is screwing me over?

ChildishGiant committed:
fix: escape ampersands so they display properly in app titles

ChildishGiant committed:
fix some offscreen/truncated elements when using smaller windows

ChildishGiant committed:
feat: make output selector actually work again

ChildishGiant committed:
update docs and fix some formatting
